nvflare/private/fed/server/job_meta_validator.py

```diff
@@ -46,14 +56,15 @@ def validate(self, job_name: str, job_data: bytes) -> Tuple[bool, str, dict]:
 
         meta = {}
         try:
-            with ZipFile(BytesIO(job_data), "r") as zf:
-                meta = self._validate_meta(job_name, zf)
-                site_list = self._validate_deploy_map(job_name, meta)
-                self._validate_app(job_name, meta, zf)
-                clients = self._get_all_clients(site_list)
-                self._validate_min_clients(job_name, meta, clients)
-                self._validate_resource(job_name, meta)
-                self._validate_mandatory_clients(job_name, meta, clients)
+            if isinstance(job_data, bytes):
+                with ZipFile(BytesIO(job_data), "r") as zf:
+                    meta = self._validate_zf(job_name, zf)
+            elif isinstance(job_data, str):
+                # job_data is a file name
+                with ZipFile(job_data, "r") as zf:
+                    meta = self._validate_zf(job_name, zf)
```

Comment:
No. The file is created temporarily and used only during the submission process. If it is removed accidentally by some external event, then just let exception happen. This won't crash our server since we catch all exceptions.

I don't think there is much we can do here.  This is what exceptions are for.